slider/tests/test_beatmap.py

```diff
@@ -195,6 +195,86 @@ def test_hit_objects_hard_rock(beatmap):
                                                     Position(x=301, y=209)]
 
 
+def test_legacy_slider_end():
+    beatmap = slider.example_data.beatmaps.miiro_vs_ai_no_scenario()
+
+    # the very first object is a slider, it ends at 1s178ms, so with a -36ms
+    # offset for the sliderend, it should be 1s142ms
+    # Position should be x=271, y=169
+    objects = beatmap.hit_objects()
+    first_slider = objects[0]
+    assert isinstance(first_slider, slider.beatmap.Slider)
+
+    expected_lazy_pos = Position(x=271, y=169)
+
+    last_point = first_slider.true_tick_points[-1]
+    # the calculation seems to include some rounding errors, any derivation of
+    # +- slider leniency is fine
+    # slider leniency is x1.2 of the circle radius, see
+    # https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Osu.Tests/
+    # TestSceneSliderFollowCircleInput.cs#L42
+
+    # This factor is to adjust how much leniency we allow for the test
+    # The sliderball itself is rather big, so scaling this down will increase
+    # the accuracy of the test's claims
+    precision_factor = 16.0
+
+    biggest_allowed_gap = (
+        slider.beatmap.circle_radius(beatmap.circle_size)
+        * 1.2 / precision_factor
+    )
```

Comment:
very skeptical about this. As far as I can tell, there shouldn't be any "rounding errors". The legacy last tick has a definitive position and we should be calculating that exactly. 

I don't see how the slider leniency factor of 1.2x has anything to do with the location of the last slider tick. Pretty sure that's just saying you can be 20% out of the slider before you sliderbreak.

Reply:
Agree, the sliderball calc has no reason to be here.

The rounding errors i am talking about are mostly about how the editor displays the sliderball and the rounding nature of osu pixels on the grid, even if the position has floating point components.

Reply:
Instead of guesstimating visually, I would prefer if the actual location of the last slider tick was confirmed via lazer.